int32 offset
int32 resistance
int32 red
int32 green
int32 blue
int32 exercise_type
bool robot_ready          #set this insted of run_robot
bool guide_z
bool system_shutdown